modules/mono/glue/GodotSharp/GodotSharp/Core/Array.cs

```diff
@@ -44,6 +44,15 @@ public Array(IEnumerable collection) : this()
                 Add(element);
         }
 
+        public Array(params object[] what) : this()
+        {
+            Resize(what.Length);
+            for (int i = 0; i < what.Length; i++)
+            {
+                this[i] = what[i];
+            }
```

Comment:
I would prefer if all these loops where replaced with an internal call. The overhead here is too big mainly because of marshalling but also due to each get/set at index being one extra internal call.
Array and Dictionary are lazy so they only do marshalling of an element when it's accessed. As such if this loop were moved to an internal call then no marshalling would be needed.